staging/src/k8s.io/cloud-provider/cloud.go

```diff
@@ -218,6 +218,11 @@ type Route struct {
 	Name string
 	// TargetNode is the NodeName of the target instance.
 	TargetNode types.NodeName
+	// EnableNodeAddresses is a feature gate for TargetNodeAddresses. If false, ignore TargetNodeAddresses.
+	// Without this, if users haven't update their cloud-provider, reconcile() will delete and create same route every time.
+	EnableNodeAddresses bool
```

Comment:
Where would this be set?

Reply:
It should be set true by specific cloud-providers when they import this new change and `CreateRoute()`.

Reply:
Why is it a per-route setting?  Wouldn't it be set once for the route controller?

Reply:
Do you suggest putting `EnableNodeAddresses` into `RouteController struct`? It seems reasonable.
Please correct me if I'm wrong. After a specifc cloud-provider updates the cloud-provider lib, if it set `EnableNodeAddresses` to be true, it will provide `TargetNodeAddresses` in `ListRoutes()`'s results. If it doesn't set `EnableNodeAddresses`, it won't provides `TargetNodeAddresses`. I think this is the expected behaviour of cloud-provider consumers.